Code Insights: Add "Remove insight from dashboard" context menu action

vovakulikov opened the pull request:
Closes https://github.com/sourcegraph/sourcegraph/issues/23065

## Test plan
- Check that the 'Remove from this dashboard' action actually removes insight from the dashboard
- Check that other context menu items haven't been affected 

## TODO 
- [x] Add 'Remove insight from this dashboard' logic
- [x] Extend UI with another context menu item
- [x] Add integration test for this context menu item, it will be resolved in a separate issue (https://github.com/sourcegraph/sourcegraph/issues/32363)

## UI review
@AlicjaSuska can you please check copies and the general flow for removing insight from the dashboard.

AlicjaSuska reviewed (approved):
thank you @vovakulikov The functionality works, I think that as far as this PR goes (adding item to the menu) all is good :)

Couple thoughts about additional issues that need to be resolved that are connected to this topic (writing here so we have the history):
- I think we should stop using the browser confirmation modal.
It looks like an error -- both for removing and deleting
- we don't have any margin between the last row of insights and the bottom of the page - we need around 40px there.
- I think we should have some kind of alert / confirmation after removing or deleting. I need to think about the placement and behaviour
- tyle the 'delete' action - add a divider above and make it red as [here](https://www.figma.com/file/38cMCA2pOScKIqzt3ZNuiz/Limited-access-mode-RFC-Private-567-%2330165-%5BWIP%5D?node-id=1153%3A13188)

vovakulikov commented:
@AlicjaSuska thanks for your review. 
- I added 1.5 rem padding for the main layout (I tried the 2rem padding which is closer to your suggestion of about 40px, but it looks too big empty space at the bottom. 1.5 rem looks good and it matches with padding-top: 1.5rem that we currently have so I added 1.5rem
- I added a divider for the delete button. About "delete" related styles for this context menu, I could implement this now in place but by this, I would have increased the tech debt around our shared components. I think we should implement variant danger inside of wildcard Menu component, I created an issue for this for frontend platform team here https://github.com/sourcegraph/sourcegraph/issues/32378
- About confirmation feedback, I've created an issue for it https://github.com/sourcegraph/sourcegraph/issues/32374
- About the toast problem I also created an issue https://github.com/sourcegraph/sourcegraph/issues/32375
